AVInput/AVInput.cpp

```diff
@@ -18,193 +18,903 @@
-    Unregister(_T("contentProtected"));
-}
+                Core::ToString((uint8_t*)&edidVec[0], size, true, edidbase64);
+            }
+            catch (const device::Exception& err) {
+                LOG_DEVICE_EXCEPTION1(std::to_string(iPort));
+            }
+            return edidbase64;
+        }
 
-uint32_t AVInput::endpoint_numberOfInputs(const JsonObject &parameters, JsonObject &response)
-{
-    LOGINFOMETHOD();
+        /**
+         * @brief This function is used to translate HDMI input hotplug to 
+         * deviceChanged event.
+         *
+         * @param[in] input Number of input port integer.
+         * @param[in] connection status of input port integer.
+         */
+        void AVInput::AVInputHotplug( int input , int connect, int type)
+        {
+            LOGWARN("AVInputHotplug [%d, %d, %d]", input, connect, type);
+
+            JsonObject params;
+            params["devices"] = getInputDevices(type);
+            sendNotify(AVINPUT_EVENT_ON_DEVICES_CHANGED, params);
+        }
 
-    bool success = false;
+        /**
+         * @brief This function is used to translate HDMI input signal change to
+         * signalChanged event.
+         *
+         * @param[in] port HDMI In port id.
+         * @param[in] signalStatus signal status of HDMI In port.
+         */
+        void AVInput::AVInputSignalChange( int port , int signalStatus, int type)
+        {
+            LOGWARN("AVInputSignalStatus [%d, %d, %d]", port, signalStatus, type);
+
+            JsonObject params;
+            params["id"] = port;
+            std::stringstream locator;
+            if (type == HDMI) {
+                locator << "hdmiin://localhost/deviceid/" << port;
+            }
+            else {
+                locator << "cvbsin://localhost/deviceid/" << port;
+            }
+            params["locator"] = locator.str();
+            /* values of dsHdmiInSignalStatus_t and dsCompInSignalStatus_t are same
+	       Hence used only HDMI macro for case statement */
+            switch (signalStatus) {
+                case dsHDMI_IN_SIGNAL_STATUS_NOSIGNAL:
+                    params["signalStatus"] = "noSignal";
+                    break;
+
+                case dsHDMI_IN_SIGNAL_STATUS_UNSTABLE:
+                    params["signalStatus"] = "unstableSignal";
+                    break;
+
+                case dsHDMI_IN_SIGNAL_STATUS_NOTSUPPORTED:
+                    params["signalStatus"] = "notSupportedSignal";
+                    break;
+
+                case dsHDMI_IN_SIGNAL_STATUS_STABLE:
+                    params["signalStatus"] = "stableSignal";
+                    break;
+
+                default:
+                    params["signalStatus"] = "none";
+                    break;
+            }
+            sendNotify(AVINPUT_EVENT_ON_SIGNAL_CHANGED, params);
+        }
 
-    auto result = numberOfInputs(success);
-    if (success) {
-        response[_T("numberOfInputs")] = result;
-    }
+        /**
```

Comment:
Comment needs to be updated.

Reply:
To maintain consistency , please add for all APIs

Reply:
Updated the comments